Kristin - Implement Promotion Eligibility Table in Questionnaire Dashboard

KH07 opened the pull request:
# Description
Implement frontend for the Questionnaire Dashboard's Promotion Eligibility Table based on the latest development branch.

## Related PRS (if any):
This frontend PR is related to the [#1549](https://github.com/OneCommunityGlobal/HGNRest/pull/1549) backend PR.

## Main changes explained:
- Create PromotionTable.jsx for displaying the table
- Added PromotionTable.css for styling layout
- Added a new route in routes.jsx to load the /promotiontable view.
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to /promotiontable
6. verify the UI
7. modify dummy members and verify the display changes accordingly

## Screenshots or videos of changes:
<img width="1540" height="994" alt="Promotion Table UI" src="https://github.com/user-attachments/assets/65bc3f83-efba-41b6-9bfc-9e567fa4fd1d" />

## Note:
The changes in this PR are identical to the changes in [#3729](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/3729), which is way behind the development branch and causes issues during rebase.

KH07 commented:
> This feature works as intended however the darkmode seems not to work on this page
> 
> https://www.loom.com/share/80e176bcfdab48b0805877410ef975ca

Hi, I just added dark mode support for this page.

akshith312 commented:
Web page works as intended and switches to dark mode as well, but in dark mode review for this week button doesn't get displayed and when hovering over specific records the display is not clear. The table content in mobile view is not displayed completely.

https://github.com/user-attachments/assets/3240fd29-d542-47f4-8337-910b66eb7af0

KH07 commented:
> The web page works as intended, but there are issues while toggling to dark mode.
> 
> 1. The review for this week button is not displayed in contrast to the dark theme.
> 2. While hovering over specific records in the dark mode contents again are not visible.
> 3. The table contents don't seem to be responsive in different layouts.
> 
>  Screen.Recording.2025-08-25.170928.mp4

Hi Akishith, the latest commits addressed the issues and provided a new look.

<img width="1527" height="994" alt="PR3851" src="https://github.com/user-attachments/assets/aab02234-4265-4002-808c-fa356364dc96" />

SudheeshTD reviewed (approved):
UI is working as expected, the Promotion Eligibility Table is displaying as expected in both light and dark modes.

vishnu-ing reviewed (approved):
The UI functionality is working good and the data is displayed from dummy data created in frontend in desktop view light mode. Dark mode doesn't work as expected and promotion table truncates in mobile view with no option to scroll.
<img width="602" height="527" alt="image" src="https://github.com/user-attachments/assets/9a5e686d-e6d4-4029-b6c7-0ca3482d42e3" />
<img width="959" height="535" alt="image" src="https://github.com/user-attachments/assets/de78b880-435b-4e0b-8208-3f58d2561869" />

abhirambj reviewed (changes requested):
I’ve finished testing the Promotion Eligibility Table PR and wanted to share my findings (my full process is shown in the attached video):

— When I click either the “Review for This Week” or “Process Promotions” buttons in dark mode—especially on my phone—the pressed (active) color is almost identical to the normal state, and both buttons look similar to each other. This makes it tough to tell which button has been activated or clicked, which can be confusing when I’m quickly toggling between them.

Please adjust the button styles so the active (clicked) state stands out more from the default, and so users can immediately see when a button’s been activated, especially on mobile and in dark mode.

— The “Back to Top” button is interfering with the UI on mobile devices. It either overlaps the table or isn’t positioned in an accessible way, which hurts the overall experience.

I recommend changing its placement or hiding it on small screens so it doesn’t cover content or look out of place.

These are both mainly styling issues, but they really matter for everyday usability, especially for anyone working on a phone or in dark mode.

Check out my video for a walkthrough and let me know if you need any more details!


https://github.com/user-attachments/assets/574a5ce6-6626-415c-9aff-3ead6b55643f

khan-zoha reviewed (changes requested):
Reviewed PR#3851. The light and dark mode works as espected. I have few suggestion for the dark mode. Due to the color of "Promotion Eligibility" is hard to read in the dark mode. Also the color of side of container class stays white  in the dark mode.

<img width="1512" height="982" alt="Screenshot 2025-09-06 at 3 01 29 AM" src="https://github.com/user-attachments/assets/64bcde2a-47eb-4f61-adb3-9cff2c3cff28" />
<img width="1512" height="982" alt="Screenshot 2025-09-06 at 3 01 12 AM" src="https://github.com/user-attachments/assets/aecaa551-18db-4922-828f-2704ddcf44a7" />

Aditya-gam commented:
The webpage works as intended. The table is loaded proper, dark mode toggles correctly, can highlight each entry when hovered.

#### Minor issues:
1. The Promotion Eligibility title is dark colored for both light and dark mode. It looks goon for light mode but it doesn't switch for dark mode.
		<img width="1512" height="774" alt="PromotionEligibilityTitle" src="https://github.com/user-attachments/assets/1ed9dc53-cec7-4823-9f1a-4488a2dcc43d" />
2. The "Review for This Week" and "Process Promotions" buttons have distinct active states in dark mode with changing button backgrounds but for light mode the only change is highlighting of the active button.
		<img width="1512" height="774" alt="LightModeForButtons" src="https://github.com/user-attachments/assets/02f13b09-3453-4c5f-be3b-151eba3a71d8" />
		<img width="1512" height="771" alt="DarkModeForButton" src="https://github.com/user-attachments/assets/5cb32ba9-2fd7-439e-adff-b899ee42f84f" />

The table is responsive for most of the screens.
https://github.com/user-attachments/assets/c534a029-a607-46a1-9f05-bb25944b8994

Aditya-gam reviewed (changes requested):
The webpage works as intended. The table is loaded proper, dark mode toggles correctly, can highlight each entry when hovered.

#### Minor issues:
1. The Promotion Eligibility title is dark colored for both light and dark mode. It looks goon for light mode but it doesn't switch for dark mode.
		<img width="1512" height="774" alt="PromotionEligibilityTitle" src="https://github.com/user-attachments/assets/1ed9dc53-cec7-4823-9f1a-4488a2dcc43d" />
2. The "Review for This Week" and "Process Promotions" buttons have distinct active states in dark mode with changing button backgrounds but for light mode the only change is highlighting of the active button.
		<img width="1512" height="774" alt="LightModeForButtons" src="https://github.com/user-attachments/assets/02f13b09-3453-4c5f-be3b-151eba3a71d8" />
		<img width="1512" height="771" alt="DarkModeForButton" src="https://github.com/user-attachments/assets/5cb32ba9-2fd7-439e-adff-b899ee42f84f" />

The table is responsive for most of the screens.
https://github.com/user-attachments/assets/c534a029-a607-46a1-9f05-bb25944b8994

Reviewed the Backend [PR #1549](https://github.com/OneCommunityGlobal/HGNRest/pull/1549#issuecomment-3268459694)

Aditya-gam reviewed (approved):
- Checked the changes made to the buttons. The buttons now have consistent color and consistent active and inactive states in both dark mode and light mode.'

- https://github.com/user-attachments/assets/e49b58cd-621b-4fb9-a6cc-7a54e0b5b95e

RitzzzZ2021 reviewed (changes requested):
The UI looks great in light mode, but the background isn’t set correctly in dark mode. Also, the buttons aren’t working, which may be related to the background issue.

<img width="1469" height="607" alt="image" src="https://github.com/user-attachments/assets/52d25aab-f131-4037-a6b9-0645726a94a5" />

<img width="1469" height="603" alt="image" src="https://github.com/user-attachments/assets/c8389706-ff15-485f-929d-a95ed266ad27" />

Prem203 commented:
The UI looks as expected in light mode but the background in dark mode is not completely dark. The page is responsive for all screen sizes and even smart phones. Also the buttons don't work on clicking, but if table and button creation were only intended then everything looks good except the minor change in the dark mode.

<img width="1890" height="870" alt="image" src="https://github.com/user-attachments/assets/5e20f301-81df-442a-8903-b2913d622e05" />

<img width="1877" height="951" alt="image" src="https://github.com/user-attachments/assets/8c7be091-a2a3-46ef-9925-43cabfae0b8a" />

<img width="841" height="741" alt="image" src="https://github.com/user-attachments/assets/43b71f94-c2ce-4f63-aae4-fdbf0426b67a" />

MeronTeweldebrhan reviewed (changes requested):
PR Review Feedback – PR #3851

Testing Outcome

✅ Navigated to /promotiontable and verified the UI.
✅ Modified dummy members — display updated accordingly.
✅ Dark mode functioning as expected.
✅ Overall UI verified.
❌ The “Review for this Week” and “Process Promotion” buttons are not responding.

<img width="1895" height="941" alt="Screenshot 2025-09-26 092334" src="https://github.com/user-attachments/assets/88eeb710-36f5-43b5-9355-c436d89e21d1" />
<img width="1880" height="974" alt="Screenshot 2025-09-26 092345" src="https://github.com/user-attachments/assets/77d75bfc-973a-4616-aeed-970c6fb01749" />

ShradhaMBhadrannavar reviewed (approved):
The UI is working as expected but the buttons are not responding.
<img width="1470" height="956" alt="Screenshot 2025-09-26 at 16 13 08" src="https://github.com/user-attachments/assets/a080282b-b718-4845-b054-6e1590b8faec" />

Stafvk commented:
###PR Review Feedback-PR #3851 

The UI is working as expected, and here are my observations

- The hover feature on each row works well
- The checkbox works well. When checking the checkbox, if the Process promotion button is greyed, it turns back to blue.
- Back to top button at the bottom works

##Questions:
Not sure of the expected behaviour of the buttons on the top

##Suggestions:

- Each of the texts in the column must be centered: Each td has to be centered.
<img width="1920" height="1080" alt="Screenshot (240)" src="https://github.com/user-attachments/assets/c088545a-acde-4506-a587-f69b202527b5" />

## Responsiveness
Works well for Phones


https://github.com/user-attachments/assets/7093ff77-52f2-4539-8b32-181f7a4779a7


##Keyboard Accessibility check
Works. The expected behavior for the checkbox is that it should work with the Spacebar, and it does.

sanjeev29 commented:
I tested the PR changes locally. The "Review For This Week" and "Process Promotions" button does nothing when clicked. The promote is not persistent when clicked, processed and refreshed. The dark mode for the table is working.
<img width="1421" height="607" alt="Screenshot 2025-10-17 at 7 58 07 PM" src="https://github.com/user-attachments/assets/851bd268-05f0-4603-bc04-c48ebded1349" />
<img width="1430" height="576" alt="Screenshot 2025-10-17 at 7 58 15 PM" src="https://github.com/user-attachments/assets/6fc2eb31-036b-4ef7-aee2-e14c90fce4ef" />

apoorvajainrp21 commented:
Conducted according to instructions: 
<img width="1397" height="772" alt="{1436FAE1-6A1E-4C4D-81DB-4254BE0BF0A7}" src="https://github.com/user-attachments/assets/d3ce6d1e-6e13-4313-b5cf-b700d51677f6" />
<img width="1545" height="736" alt="{AC277467-20EA-4D0B-83EF-AD9E7A800383}" src="https://github.com/user-attachments/assets/fbb6373c-c636-42c6-8a2f-b946fa4a7917" />

Vinay944924 commented:
Hi Kristin,
I checked the PR locally and checked the promotion eligibility.
it worked fine in the light mode but when it is turned into dark mode it is not working only the promotion table is converted to dark mode but not the whole screen. Please review it before you make any changes. I have provided the screenshot below.
<img width="1470" height="878" alt="Screenshot 2025-10-31 at 1 51 49 PM" src="https://github.com/user-attachments/assets/aa295ae8-9705-47a0-9a9e-ce5c84c8d3c0" />

Sriamshreddy000 commented:
Tested this PR (Frontend [#3851](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/3851)) along with Backend PR [#1549](https://github.com/OneCommunityGlobal/HGNRest/pull/1549).
The frontend setup ran fine, but the backend ran into build issues related to the sharp dependency.
It appears to be a compatibility issue on ARM64 (Apple Silicon) systems.
Left my detailed feedback and error screenshots in the backend PR [#1549](https://github.com/OneCommunityGlobal/HGNRest/pull/1549).

SwathiAngadi commented:
Hi Kristin,

Promotion Table UI loads perfectly , table is responsive too. Dark mode works as expected. Hover highlights the selected row. Review for this week and Process Promotion buttons are just displayed no functionalities added as of now. However I was not able to modify the records with admin/owner  except clicking checkboxes but it is not  persistent on page refresh. 
<img width="1903" height="940" alt="Screenshot 2025-12-08 221921" src="https://github.com/user-attachments/assets/2da638f7-5d2d-43e9-898f-9ae636b2a49e" />
<img width="1911" height="943" alt="Screenshot 2025-12-08 222610" src="https://github.com/user-attachments/assets/be221f11-9946-450f-be9d-4694d4748b44" />
